src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -126,6 +126,13 @@ public async Task FindAsync<TArg>(
         if (searchingChecksumsLeft.Count == 0)
             return;
 
+        if (assetPath.IncludeSolutionSynchronization)
+        {
+            // Wait for the workspace to finish synchronizing changes before continuing
+            var workspace = compilationState.Services.GetRequiredService<IWorkspaceAccessorService>().Workspace;
+            await workspace.WaitForScheduledTasksAsync().ConfigureAwait(false);
```

Comment:
this seems like this is gating this sync working on current workspace work completing (which might be large amounts of *unrelated* work).

Reply:
This only waits for workspace event handler callbacks, which _should_ be executing quickly. It doesn't wait for second-level operation queues.

Reply:
my preference would be to wait for just the operations that send the text over.  it's all a bit to disconnected here, meaning it's just too easy to break this.  I would prefer (if we go this route) to have very tight coupling.  The sync process says: "ensure pending text changes are sync'ed over first" then it does its work.

it would also be fine if the *outgoing* call system for features did this.  So any outgoing rpc call (that specified a Solutino/Proj to sync) had that call wait on the text change sync.